nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -1463,7 +1463,12 @@ contract StakingEscrow is Issuer, IERC900History {
                 subStakeInfoToCheck.lockedValue == subStakeInfo.lockedValue);
         }
 
-        // TODO: add here validation of history (need help from @vzotova)
+        // it's not perfect because checks not only slot value but also decoding
+        // at least without additional functions
+        require(delegateGet(_testTarget, this.totalStakedForAt.selector, staker, bytes32(block.timestamp)) ==
+            totalStakedForAt(stakerAddress, block.timestamp));
+        require(delegateGet(_testTarget, this.totalStakedAt.selector, bytes32(block.timestamp)) ==
+            totalStakedAt(block.timestamp));
```

Comment:
We use `block.number` instead of the timestamp

Reply:
✔️